packages/core/src/checks.ts

```diff
@@ -0,0 +1,1274 @@
+        code: "not_multiple_of",
+        divisor: def.value as number,
+        input: payload.value,
+        inst,
+        continue: !def.abort,
+      });
+    };
+  });
+
+/////////////////////////////////////
+/////    $ZodCheckFinite    /////
+/////////////////////////////////////
+// interface $ZodCheckFiniteDef extends $ZodCheckDef {
+//   check: "finite";
+// }
+
+// export interface $ZodCheckFinite extends $ZodCheckInternals<number> {
+//   _def: $ZodCheckFiniteDef;
+//   _issc:
+//     | errors.$ZodIssueTooBig<"number", number>
+//     | errors.$ZodIssueTooSmall<"number", number>;
+// }
+
+// export const $ZodCheckFinite: core.$constructor<$ZodCheckFinite> =
+//   core.$constructor("$ZodCheckFinite", (inst, def) => {
+//     $ZodCheck.init(inst, def);
+
+//     inst._zod.onattach = (inst) => {
+//       inst["_computed"].finite = true;
+//     };
+
+//     inst._zod.check = (payload) => {
+//       if (Number.isFinite(payload.value)) return;
+//       payload.issues.push({
+//         origin: "number",
+//         ...(payload.value === Number.POSITIVE_INFINITY
+//           ? {
+//               code: "too_big",
+//               maximum: Number.POSITIVE_INFINITY,
+//             }
+//           : {
+//               code: "too_small",
+//               minimum: Number.NEGATIVE_INFINITY,
+//             }),
+//         // code: payload.value === Number.POSITIVE_INFINITY ? "too_big" : "too_big",
+//         // maximum: Number.POSITIVE_INFINITY,
+//         inclusive: false,
+//         input: payload.value,
+//         inst,
+//       });
+//     };
+//   });
+
+///////////////////////////////////////
+/////    $ZodCheckNumberFormat    /////
+///////////////////////////////////////
+
+export type $ZodNumberFormats = "int32" | "uint32" | "float32" | "float64" | "safeint";
+
+export interface $ZodCheckNumberFormatDef extends $ZodCheckDef {
+  check: "number_format";
+  format: $ZodNumberFormats;
+  // abort?: boolean;
+}
+
+export interface $ZodCheckNumberFormatInternals extends $ZodCheckInternals<number> {
+  def: $ZodCheckNumberFormatDef;
+  issc: errors.$ZodIssueInvalidType | errors.$ZodIssueTooBig<"number"> | errors.$ZodIssueTooSmall<"number">;
+}
+
+export interface $ZodCheckNumberFormat extends $ZodCheck<number> {
+  _zod: $ZodCheckNumberFormatInternals;
+}
+
+export const $ZodCheckNumberFormat: core.$constructor<$ZodCheckNumberFormat> = /*@__PURE__*/ core.$constructor(
+  "$ZodCheckNumberFormat",
+  (inst, def) => {
+    $ZodCheck.init(inst, def); // no format checks
+    def.format = def.format || "float64";
+
+    const isInt = def.format?.includes("int");
+    const origin = isInt ? "int" : "number";
+    const [minimum, maximum] = util.NUMBER_FORMAT_RANGES[def.format!];
+
+    inst._zod.onattach = (inst) => {
+      inst._zod.computed.format = def.format;
+      inst._zod.computed.minimum = minimum;
+      inst._zod.computed.maximum = maximum;
+      if (isInt) inst._zod.computed.pattern = regexes.integer;
+    };
```

Comment:
I noticed that limits are not set correctly for `z.number().int()`, because `inclusive: true` is not set here

<img width="677" alt="Screenshot 2025-04-17 at 09 21 53" src="https://github.com/user-attachments/assets/7986573c-6c74-4885-ae48-1c8d132d5183" />

Reply:
Also noticed that when `minimum` is set to `0`, then it's not printed at all by `toJsonSchema`

Reply:
Possible fix https://github.com/colinhacks/zod/pull/4224